ethers-providers/src/rpc/pubsub.rs

```diff
@@ -94,16 +94,18 @@ where
             return Poll::Ready(next_element)
         }
 
-        let this = self.project();
-        match futures_util::ready!(this.rx.poll_next(ctx)) {
-            Some(item) => match serde_json::from_str(item.get()) {
-                Ok(res) => Poll::Ready(Some(res)),
-                Err(err) => {
-                    error!("failed to deserialize item {:?}", err);
-                    Poll::Pending
-                }
-            },
-            None => Poll::Ready(None),
+        let mut this = self.project();
+        loop {
```

Comment:
isn't this loop a no-op, as the return always ends it on the first loop?

Reply:
nvm i am misreading the code

Reply:
This continues the poll `rx` on `Err(err)` instead of returning Pending, and ensures that the underlying stream is completely drained or returns a valid element.